boards/arm/ip_k66f/board.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2020 DENX Software Engineering GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <drivers/gpio.h>
+
+static int ip_k66f_board_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	struct device *gpioa =
+		device_get_binding(DT_LABEL(DT_NODELABEL(gpioa)));
+
+	/* Set !ETH_RST */
+	gpio_pin_configure(gpioa, 24, GPIO_OUTPUT_HIGH);
```

Comment:
Another way of handling this could be to have a reset GPIO for the PHY and have the ethernet driver bring the PHY out of reset before use. With the current approach, the PHY will be taken out of reset regardless of whether ethernet is to be used (I know, on this board you probably have more use-cases with ethernet enabled than without, but still).

Reply:
I would also recommend to get this information from the device tree file and handle reset in the Ethernet driver itself.. This would increase the portability of the driver somewhat (next time no one will have to write a board level device tree file for PHY reset)..

Reply:
Could you be more specific about the "Ethernet" driver? Would you recommend adding the generic PHY reset code into eth_mcux.c (which is NXP's Kinetics specific) or maybe to ethernet.c's ethernet_init() function (which looks to be more generic)?

Reply:
I will drop this code - i.e. the board.c file from this patch set. Instead I will introduce new PR and patches to address this topic (as it seems to be more generic than fixes for this particular board).

Reply:
But I would still like to hear your recommendation if this reset code shall be put into eth_mcux.c or ethernet.c ?

Reply:
The subsys/net/l2/ethernet/ethernet.c is for generic, hw agnostic L2 code. This means that it is not suppose to have any gpio configure calls. So all of this kind of code should stay inside drivers/ethernet directory. You can create some common file there if you think more than one driver would benefit from such calls.

Reply:
We should do something similar to how the linux kernel handles ethernet phy, and ethernet phy resets. 
Please see https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/net/ethernet-phy.yaml#L141 which defines the "reset-gpios" property for Ethernet PHYs. 

As this is generic across all Ethernet PHYs, it should be placed in a common file and there should be a global function (accessible by all Ethernet drivers) to optionally call up this function at boot up. 

You can take https://patchwork.kernel.org/patch/7871481/ as reference which adds support for "reset-gpios" binding in the cadence gem driver (AKA macb)..

Reply:
Ok, thanks for the input. It turns out that HW on the ip_k66f board resets the PHY, so there is no urgent need for such code.
I've already dropped the board.c from this patch set and will introduce it when needed.